fineract-provider/src/main/java/org/apache/fineract/infrastructure/core/service/migration/TenantPasswordEncryptionTask.java

```diff
@@ -0,0 +1,90 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.fineract.infrastructure.core.service.migration;
+
+import java.sql.ResultSet;
+import java.sql.Statement;
+import java.util.Optional;
+import liquibase.change.custom.CustomTaskChange;
+import liquibase.database.Database;
+import liquibase.database.jvm.JdbcConnection;
+import liquibase.exception.CustomChangeException;
+import liquibase.exception.SetupException;
+import liquibase.exception.ValidationErrors;
+import liquibase.resource.ResourceAccessor;
+import org.apache.fineract.infrastructure.security.utils.EncryptionUtil;
+
+public class TenantPasswordEncryptionTask implements CustomTaskChange {
+
+    private final String encryptionType;
+    private final String masterKey;
+
+    public TenantPasswordEncryptionTask() {
+        this.encryptionType = Optional.ofNullable(System.getenv("FINERACT_DEFAULT_TENANTDB_ENCRYPTION"))
```

Comment:
Unfortunately this is not enough. The properties can be overridden directly via program arguments for example so this is not going to be in an environment variable necessarily.
I know this object will be managed by Liquibase so no Spring DI at this point but you can do the ugly way to do it, look at TemporaryConfigurationServiceContainer